includes/wc-core-functions.php

```diff
@@ -722,19 +722,32 @@ function wc_get_image_size( $image_size ) {
 		'crop'   => 1,
 	);
 
+	// Size mapping.
+	if ( in_array( $image_size, array( 'shop_single', 'woocommerce_single' ), true ) ) {
+		$image_size = 'single';
+	} elseif ( in_array( $image_size, array( 'shop_thumbnail', 'shop_catalog', 'woocommerce_thumbnail' ), true ) ) {
+		$image_size = 'thumbnail';
+	}
+
 	if ( is_array( $image_size ) ) {
 		$size       = array(
 			'width'  => isset( $image_size[0] ) ? absint( $image_size[0] ) : 600,
 			'height' => isset( $image_size[1] ) ? absint( $image_size[1] ) : 600,
 			'crop'   => isset( $image_size[2] ) ? absint( $image_size[2] ) : 1,
 		);
 		$image_size = $size['width'] . '_' . $size['height'];
-	} elseif ( in_array( $image_size, array( 'single', 'shop_single', 'woocommerce_single' ), true ) ) {
+
+	} elseif ( 'single' === $image_size ) {
 		$size['width']  = absint( wc_get_theme_support( 'single_image_width', get_option( 'woocommerce_single_image_width', 600 ) ) );
 		$size['height'] = '';
 		$size['crop']   = 0;
-		$image_size     = 'single';
-	} elseif ( in_array( $image_size, array( 'thumbnail', 'shop_thumbnail', 'shop_catalog', 'woocommerce_thumbnail' ), true ) ) {
+
+	} elseif ( 'gallery_thumbnail' === $image_size ) {
+		$size['width']  = absint( wc_get_theme_support( 'gallery_thumbnail_image_width', 100 ) );
```

Comment:
This defaults to 100, but if we want to take advantage of using WP core thumbnail if exists, shouldn't it default to 150 x 150?

Reply:
Because of the way it's written, if there is no 100 thumbnail, it will use the nearest (150 if present).